transformations/propbank_srl_roles/README.md

```diff
@@ -0,0 +1,67 @@
+# SRL arguments exchange perturbation 🦎  + ⌨️ → 🐍
+This perturbation adds noise to all types of text sources (sentence, paragraph, etc.) proportional to noise erupting
+from keyboard typos making common spelling errors.
+
+Author name: Priyank Soni
+Author email: priyank.soni@ipsoft.com
+Author Affiliation: Amelia, Ipsoft
+
+## What type of a transformation is this?
+This transformation acts like a perturbation to test robustness. First we are getting semantic role labels with SRL BERT which is state of the art single model for English PropBank SRL.  Few letters picked at random are replaced with letters
+which are at keyboard positions near the source letter. Generated transformations display high similarity to the
+source sentences i.e. the code outputs highly precise generations.
+
+Then we apply following rules to modify sentence, so that semantically they remain correct:
+
+1. if ARGM-LOC and ARGM-TMP both present, exchange them.
+
```

Comment:
There are other PropBank "modifier" arguments too which could be used here for exchanging: PNC (purpose), MNR (manner), cause, and DIR (direction)?

Reply:
@kaustubhdhole, thank you for the suggestions
 
Will add these modifications also:

1. if ARG0 and ARGM-PRP are both present and ARGM-PRP comes after ARG0, put ARGM-PRP just before ARG0.

Example: We will be going to the office for future meetings

[ARG0: We] [ARGM-MOD: will] be [V: going] [ARG4: to the office] [ARGM-PRP: for future meetings] . --> For future meetings, we will be going to the office.

2. if ARG0 and ARGM-CAU are both present and ARGM-CAU comes after ARG0, put ARGM-CAU just before ARG0.

Example: I am going home because it's raining.

[ARG0: I] am [V: going] [ARG4: home] [ARGM-CAU: because it 's raining] . -->  Because it's raining I am going home.

3.  if ARG0 and ARGM-GOL are both present and ARGM-GOL comes after ARG0, put ARGM-GOL just before ARG0.
 
Example: He played the match for his brother.

[ARG0: He] [V: played] [ARG1: the match] [ARGM-GOL: for his brother] . --> For his brother He played the match.

4. if ARGM-TMP and ARGM-PRP both present, exchange them.

Example: From 2023 we will be playing football for fitness.

[ARGM-TMP: From 2023] [ARG0: we] [ARGM-MOD: will] be [V: playing] [ARG1: football] [ARGM-PRP: for fitness] . -->   F or fitness we will be playing football from 2023.

5.  if ARGM-TMP and ARGM-CAU both present, exchange them.

Example: In June the team will not practice because of the rain.

[ARGM-TMP: In June] [ARG0: the team] [ARGM-MOD: will] [ARGM-NEG: not] [V: practice] [ARGM-CAU: because of the rain] . -->  Because of the rain the team will not practice In June.  


But it will increase the size of the dataset very much as a particular sentence may fit in more than one of the above rules.

Reply:
Thanks @priyanksonis - these look great! It is okay to over-generate and then prune later. The best way to make this usable is if these are kept configurable the constructor level.